cmd/dockerd/daemon_windows.go

```diff
@@ -93,7 +94,30 @@ func newCgroupParent(config *config.Config) string {
 	return ""
 }
 
-func (cli *DaemonCli) initContainerD(_ context.Context) (func(time.Duration) error, error) {
+func (cli *DaemonCli) initContainerD(ctx context.Context) (func(time.Duration) error, error) {
+	if cli.Config.ContainerdAddr == "" && cli.Config.DefaultRuntime == config.WindowsV2RuntimeName {
+		cli.Config.ContainerdAddr = supervisor.GrpcPipeName
+		system.InitContainerdRuntime(cli.Config.ContainerdAddr)
```

Comment:
^^ related to my previous comment; if we set it as part of `setDefaults()`, then this could be moved to the end of this function, and use `r.Address()`


With that, we also wouldn't have to export `supervisor.GrpcPipeName`